eis_toolkit/statistical_analyses/statistical_tests.py

```diff
@@ -0,0 +1,123 @@
+import pandas as pd
+from beartype import beartype
+from beartype.typing import Literal, Optional, Sequence, Tuple
+from scipy.stats import chi2_contingency, shapiro
+
+from eis_toolkit import exceptions
+
+
+@beartype
+def check_empty_dataframe(data: pd.DataFrame):
+    """Check if the input dataframe is empty.
+
+    Args:
+        data: Input DataFrame
+
+    Raises:
+        EmptyDataFrameException: The input DataFrame is empty.
+    """
+    if data.empty:
+        raise exceptions.EmptyDataFrameException("The input DataFrame is empty.")
+
+
+@beartype
+def chi_square_test(data: pd.DataFrame, target_column: str) -> Sequence[Tuple[float, float, int]]:
+    """Compute Chi-square test for independence on categorical data.
+
+    Args:
+        data: DataFrame containing the input data.
+        target_column: Variable against which independence of other variables is tested.
+
+    Raises:
+        InvalidParameterValueException: The target_column is not in input DataFrame.
+
+    Returns:
+        Test statistics for each variable (except target_column).
```

Comment:
Assumptions of chi2 could be listed in the docstring. Implementation looks overall good, but rather than listing all the results (which can be many if a DF has many columns), the results could be put into a dictionary where the key can detail which colum was used. Also, thinking how this might be called from the plugin, maybe letting the user the specify the included columns is a good idea (as many as they want, and if they leave it None/empty, all columns are included).